### PR TITLE
WFPREV-145 - Initial Rest endpoints including projects

### PR DESCRIPTION
**Pull request by agordon-vivid**
This includes REST CRUD for projects, program areas, code tables, project boundaries.

### CODE REVIEW

File: server/wfprev-api/src/main/java/ca/bc/gov/nrs/wfprev/common/serializers/GeoJsonJacksonDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      log.trace("<deserialize");
+      Geometry result = null;
+      ObjectCodec oc = jsonParser.getCodec();
+      JsonNode node = (JsonNode)oc.readTree(jsonParser);
+      String geoJson = node.toString();
+
+      try {
+         Reader reader = new StringReader(geoJson);
+         result = geoJsonReader.read(reader);
+      } catch (ParseException var9) {
```
Comment (ssylver93) [inline server/wfprev-api/src/main/java/ca/bc/gov/nrs/wfprev/common/serializers/GeoJsonJacksonDeserializer.java:43]: What is the purpose of this 'var9' 'var10' naming convention?

Comment (agordon-vivid) [reply server/wfprev-api/src/main/java/ca/bc/gov/nrs/wfprev/common/serializers/GeoJsonJacksonDeserializer.java:43]: none.  I hadn't noticed it.  Probably got added by somebody's IDE automatically.  I'll change them to 'e' for the exception convention

File: server/wfprev-api/src/main/java/ca/bc/gov/nrs/wfprev/common/serializers/GeoJsonJacksonSerializer.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+   public void serialize(Geometry geometry, JsonGenerator generator, SerializerProvider provider) throws IOException, JsonProcessingException {
+      log.trace("<serialize");
+
+      try {
+         GeoJsonWriter geoJsonWriter = new GeoJsonWriter();
+         StringWriter writer = new StringWriter();
+         geoJsonWriter.write(geometry, writer);
+         String json = writer.toString();
+         generator.writeRawValue(json);
+      } catch (RuntimeException var7) {
```
Comment (ssylver93) [inline server/wfprev-api/src/main/java/ca/bc/gov/nrs/wfprev/common/serializers/GeoJsonJacksonSerializer.java:31]: 'var7' Here also

Comment (agordon-vivid) [reply server/wfprev-api/src/main/java/ca/bc/gov/nrs/wfprev/common/serializers/GeoJsonJacksonSerializer.java]: ya, i caught it.  SHould be gone now

File: terraform/autoscaling.tf [hunk 3]
```diff
@@ -95,98 +95,98 @@ resource "aws_cloudwatch_metric_alarm" "wfprev_service_cpu_low" {
...
+#      ClusterName = aws_ecs_cluster.wfprev_main.name
+#      ServiceName = aws_ecs_service.client.name
+#    }
+
+#    alarm_actions = [aws_appautoscaling_policy.wfprev_client_up.arn]
+
+#    tags = {
+#      Environment = "${var.TARGET_ENV}"
+#    }
+#  }
```
Comment (ssylver93) [inline terraform/autoscaling.tf:192]: I take it load balancing and autoscaling is not needed now that the UI is handled in CloudFront and S3? Are they something that comes with CloudFront by default or do we need to do more work for that?

**Review by ssylver93 — APPROVED** [review]
Looks great, just left a couple of comments/questions on areas I'm not overly clear on.

**Review by ssylver93 — APPROVED** [review]

**Review by dhlevi — DISMISSED** [review]

**Review by yzlucas — DISMISSED** [review]

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -34,7 +34,7 @@ jobs:
       - name: Change directory to Angular app
         run: cd client/wfprev-war/src/main/angular
 
-      # Use the NPMRC secret to set up npm authentication
+      # Use the NPMRC secret to set up npm authenticationq
```
Comment (ssylver93) [inline .github/workflows/test.yml:37]: Typo

File: client/wfprev-war/src/main/angular/.github/test.yml [hunk 5]
```diff
@@ -21,7 +21,7 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
 
-      - run: npm install
+      - run: npm install --legacy-peer-deps
```
Comment (ssylver93) [inline client/wfprev-war/src/main/angular/.github/test.yml:24]: Don't we want --legacy-peer-deps removed?

**Review by ssylver93 — CHANGES_REQUESTED** [review]
Looks great. I left a couple of comments for typos and the use of --legacy-peer-deps.

**Review by ssylver93 — APPROVED** [review]

**Pull request merged**